### PR TITLE
Implement locked rays in Interactive Graph

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Locked rays were not implemented along with lines. But here they are!

I used the movable ray for inspiration for the implementation. This is
using the "in-house" Vector component rather than than Mafs Vector.

Note that the ray arrow cannot be hidden/shown. This will likely be
done as part of LEMS-1929.

Issue: https://khanacademy.atlassian.net/browse/LEMS-1928

## Test plan:
`yarn jest packages/perseus/src/widgets/__tests__/interactive-graph.test.ts`

Relevant storybook stories
http://localhost:6006/?path=/story/perseus-editor-widgets-interactive-graph-editor--with-locked-lines
http://localhost:6006/?path=/story/perseus-widgets-interactive-graph--segment-with-mafs-and-locked-lines
http://localhost:6006/?path=/story/perseus-widgets-interactive-graph--all-locked-rays
http://localhost:6006/?path=/docs/perseus-editor-editorpage--docs

## Storybook screenshots

| Interactive Graph | Interactive Graph Editor | Editor Page |
|---|---|---|
| Screenshot 2024-04-23 at 6 19 46 PM | Screenshot 2024-04-23 at 6 23 22 PM | Screenshot 2024-04-23 at 6 23 50 PM |

### CODE REVIEW

File: packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap
Comment (mark-fitzgerald) [inline packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap:4188]: Is `undefined` expected?

File: packages/perseus/src/widgets/interactive-graphs/graphs/components/svg-line.tsx
Comment (nishasy) [inline packages/perseus/src/widgets/interactive-graphs/graphs/components/svg-line.tsx:23]: This was making svg `<line>` have `class="undefined"`. Now it should have no class name if one isn't passed in.

File: packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap
Comment (nishasy) [reply packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap:4188]: Good catch! Fixed.

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx:11]: minor: this type is a bit confusing. "The range is a pair of `Range`s". I might prefer to use the `Interval` type from Mafs, and label the elements of the pair:

```ts
import type {Interval} from "mafs";
// ...
type Props = {
  // ...
  range: [x: Interval, y: Interval];
}
```

File: packages/perseus/src/widgets/__tests__/interactive-graph.test.ts
Comment (benchristel) [inline packages/perseus/src/widgets/__tests__/interactive-graph.test.ts:364]: This selector seems brittle - could we use a different class for locked rays? E.g. `.locked-ray`?

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx:27]: Could we move the declaration of `otherProps` inside the `map` callback, next to Figure?

```ts
{lockedFigures.map((figure, index) => {
    let Figure;
    let otherProps;
```

The way it is now, the `otherProps` will persist between calls to the `map` callback. This means that if the first element in `lockedFigures` is a line, and the second is a point, the `range` prop will get mistakenly added to the point. Probably not a huge deal, but could create headaches down the line.

I'm also kind of surprised TypeScript accepts this, and I wonder if it's just throwing up its hands and implicitly typing `otherProps` as `any` or something.

An alternative would be to render the figure inside the `switch`:

```tsx
{lockedFigures.map((figure, index) => {
    switch (figure.type) {
        case "point":
            return <LockedPoint key={`point-${index}`} {...figure} />;
            break;
        case "line":
            return <LockedLine key={`line-${index}`} range={props.range} {...figure} />;
            break;
        default:
            /**
             * Devlopment-time future-proofing: This should
             * fail during type-checking if we add a new locked
             * shape type and forget to handle it in any other
             * switch case here.
             */
            throw new UnreachableCaseError(figure);
    }
})}
```

**Review by jeremywiebe — COMMENTED** [review]
This is so exciting to see these locked features coming through!  

image

@benchristel and @mark-fitzgerald did you have any reservations that you saw were blocking approving this PR once your comments were resolved?

**Review by mark-fitzgerald — APPROVED** [review]
Looks good. I like Ben's points, especially the one about the `otherProps` in `graph-locked-layer.tsx`.

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx
Comment (nishasy) [reply packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx:27]: oh yeah rendering directly inside the `swtich` makes way more sense. The current implementation is left over from how it was done pre-refactor where it was super long awkward to put in a swtich. Thanks!

File: packages/perseus/src/widgets/__testdata__/interactive-graph.testdata.ts
Comment (nishasy) [reply packages/perseus/src/widgets/__testdata__/interactive-graph.testdata.ts:1708]: That's a good idea. I'm going to look into it in a new PR.

File: packages/perseus/src/widgets/__stories__/interactive-graph.stories.tsx
Comment (nishasy) [reply packages/perseus/src/widgets/__stories__/interactive-graph.stories.tsx:132]: (talked offline)
Since this is in a different package, I'll hold off on doing that. If I keep adding to this file, I'll move the flags so they can be used in perseus though.

File: packages/perseus/src/widgets/interactive-graphs/locked-line.tsx
Comment (nishasy) [reply packages/perseus/src/widgets/interactive-graphs/locked-line.tsx:48]: I don't think our in-house Vector (build offs of SVGLine) has solid/dashed style option? I'm not sure it would work the same way?

**Review by benchristel — APPROVED** [review]
LGTM!

**Pull request merged**
